package.json

```diff
@@ -41,44 +41,43 @@
     "reflux": "^0.4.1"
   },
   "devDependencies": {
-    "babel-core": "^6.22.1",
-    "babel-loader": "^6.2.10",
-    "babel-plugin-syntax-flow": "^6.8.0",
-    "babel-plugin-transform-flow-strip-types": "^6.8.0",
-    "babel-preset-es2015": "^6.22.0",
-    "babel-preset-react": "^6.22.0",
-    "babel-preset-stage-0": "^6.22.0",
-    "css-loader": "^0.27.3",
-    "exports-loader": "^0.6.3",
-    "file-loader": "^0.10.0",
-    "grunt": "~1.0.1",
-    "grunt-contrib-clean": "~1.0.0",
-    "grunt-contrib-connect": "~1.0.2",
-    "grunt-contrib-copy": "~1.0.0",
-    "grunt-open": "~0.2.3",
-    "grunt-webpack": "^2.0.1",
-    "http-server": "^0.9.0",
+    "@babel/core": "^7.1.2",
+    "@babel/plugin-proposal-class-properties": "^7.0.0",
+    "@babel/plugin-syntax-flow": "^7.0.0",
+    "@babel/plugin-transform-flow-strip-types": "^7.0.0",
+    "@babel/preset-env": "^7.1.0",
+    "@babel/preset-react": "^7.0.0",
+    "babel-loader": "^8.0.0",
+    "css-loader": "^1.0.0",
+    "exports-loader": "^0.7.0",
+    "file-loader": "^2.0.0",
+    "grunt": "^1.0.3",
+    "grunt-contrib-clean": "^2.0.0",
+    "grunt-contrib-connect": "^2.0.0",
+    "grunt-contrib-copy": "^1.0.0",
+    "grunt-force-task": "^2.0.0",
+    "grunt-open": "^0.2.3",
+    "grunt-webpack": "^3.1.3",
+    "http-server": "^0.11.1",
     "idb-pull-blob-store": "^0.5.1",
-    "json-loader": "^0.5.4",
     "libp2p-crypto-secp256k1": "^0.1.2",
-    "load-grunt-tasks": "~3.5.0",
-    "node-sass": "^4.5.0",
-    "normalize.css": "~5.0.0",
+    "load-grunt-tasks": "^4.0.0",
+    "node-sass": "^4.3.3",
+    "normalize.css": "^8.0.0",
     "ping-ops-requests": "~1.0.4",
-    "react-hot-loader": "^1.3.0",
-    "sass-loader": "^6.0.3",
-    "style-loader": "~0.15.0",
-    "transform-loader": "^0.2.3",
-    "uglify-js": "github:mishoo/UglifyJS2#harmony",
-    "url-loader": "~0.5.7",
-    "webpack": "^2.3.0",
-    "webpack-dev-server": "^2.2.1"
+    "react-hot-loader": "^4.3.11",
+    "sass-loader": "^7.1.0",
+    "style-loader": "^0.23.0",
+    "url-loader": "^1.1.1",
+    "webpack": "^4.20.2",
+    "webpack-cli": "^3.1.2",
+    "webpack-dev-server": "^3.1.9"
   },
   "scripts": {
     "test": "echo \"Error: no test specified\" && exit 1",
     "dev": "grunt serve",
     "build": "npm run build:dist",
-    "build:dist": "grunt build && webpack --config config/webpack.config.dist.js --hide-modules --progress --env production && uglifyjs dist/assets/app.js -o dist/assets/app.js",
+    "build:dist": "grunt build && webpack --config config/webpack.config.dist.js --hide-modules --progress",
```

Comment:
Did I understand your comment in https://github.com/orbitdb/orbit-web/pull/21#issue-219681164 correctly that webpack now runs in "production" mode by default?

Reply:
Yeah, if no "mode" has been set then webpack 4 will run in "production" mode by default. But at the same time I am explicitly setting the mode to "production" in the `webpack.config.dist.js` config file. The mode can be set on the command line as well but I thought it would be cleaner if "dist" config represented a "production" build. I might be wrong with this assumption.

Note that I am also removing the `webpack.DefinePlugin` (which sets the `process.env.NODE_ENV`) since that is now done automatically by Webpack. More on this here: [webpack mode/#usage](https://webpack.js.org/concepts/mode/#usage)

Reply:
Got it 👍 Makes sense and you're correct with your assumption that `dist` == `production` (as in: minified build).